src/Controls/src/Build.Tasks/XamlCache.cs

```diff
@@ -38,8 +39,13 @@ public TypeDefinition Resolve(TypeReference typeReference) =>
 	public FieldReference GetOrAddFieldReference((ModuleDefinition module, string fieldRefKey) key, Func<(ModuleDefinition module, string fieldRefKey), FieldReference> valueFactory) =>
 		GetOrAdd(_fieldReferenceCache, key, valueFactory);
 
-	public TypeReference GetOrAddTypeReference(ModuleDefinition module, (string assemblyName, string clrNamespace, string typeName) type) =>
-		GetOrAdd(_typeReferenceCache, (module, type.ToString()), x => x.module.ImportReference(x.module.GetTypeDefinition(this, type)));
+	public TypeReference GetOrAddTypeReference(ModuleDefinition module, (string assemblyName, string clrNamespace, string typeName) type) => GetOrAdd(_typeReferenceCache, (module, type.ToString()), x =>
+	{
+		if (type.typeName.EndsWith("[]"))
```

Comment:
While `type.typeName.EndsWith("[]") `works, EndsWith internally traverses the string. Not sure if we have types with less than 2 characters but, consider checking its length first:

`if (type.typeName.Length >= 2 && type.typeName.EndsWith("[]"))`